### PR TITLE
Remove cache machine

### LINKED ISSUE #7743: Remove django-cache-machine
After quite some time we finally decided to remove django-cache-machine.

It's partially blocking our Django 1.11 upgrade (#5271) and I assume there will be a measurable positive impact from removing it.

This issue is to track what's needed to be done to get there.

TODO:

* [x] Remove all `no_cache()` calls
* [x] Do not inherit from `CachingQuerySet`, `CachingManager`, or `CachingMixin`
* [x] Backport Django 1.11's `cache.get_or_set` instead of our own `cached` helper which is similar but less tested. Makes upgrading easier later.
* [x] Make sure `assert_cache_requests` can use cache-aliases

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Fixes #7743

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
As per my recent email, let's land this **after** our next tag so that a deployment can be targeted on July 5th.

File: src/olympia/addons/views.py [hunk 1]
```diff
@@ -141,14 +140,6 @@ def extension_detail(request, addon):
         return render(request, 'addons/impala/details.html', ctx)
 
 
-def _category_personas(qs, limit):
-    def fetch_personas():
-        return randslice(qs, limit=limit)
-    # TODO: .query_key comes from cache-machine, find replacement
-    key = make_key('cat-personas:' + qs.query_key(), normalize=True)
-    return cache_get_or_set(key, fetch_personas)
```
Comment (diox) [inline src/olympia/addons/views.py]: Are you sure we can remove this caching? The query is nasty.

Comment (EnTeQuAk) [reply src/olympia/addons/views.py]: I may optimize this a bit, I'm testing this currently a bit more in-depth. I'm testing the whole view though so the solution might be a bit more than just caching this query.

Comment (EnTeQuAk) [reply src/olympia/addons/views.py]: tldr; this is the first patch removing cache-machine, more performance optimizations may be added in later PRs. That part on it's own doesn't add too much more stress on the database from what I found, can be optimized though (the whole view can)

File: src/olympia/amo/decorators.py [hunk 2]
```diff
@@ -129,7 +111,7 @@ def wrapper(*args, **kw):
 
 
 def write(f):
-    return use_master(skip_cache(f))
+    return use_master(f)
```
[the remark this answers is not in the document]
Comment (EnTeQuAk) [reply src/olympia/amo/decorators.py]: Filed https://github.com/mozilla/addons-server/issues/8644

File: src/olympia/amo/models.py [hunk 3]
```diff
@@ -43,17 +32,6 @@ def use_master():
...
-    """Within this context, no queries come from cache."""
-    old = getattr(_locals, 'skip_cache', not settings.CACHE_MACHINE_ENABLED)
-    _locals.skip_cache = True
-    try:
-        yield
-    finally:
-        _locals.skip_cache = old
-
-
 class TransformQuerySet(queryset_transform.TransformQuerySet):
```
[the remark this answers is not in the document]
Comment (EnTeQuAk) [reply src/olympia/amo/models.py]: Fixed in d793bc96d5

File: src/olympia/addons/views.py (hunk 1 above)
Comment (EnTeQuAk) [reply src/olympia/addons/views.py]: I added it back as suggested in 72664b17aa

File: src/olympia/amo/models.py [hunk 4]
```diff
@@ -94,20 +67,14 @@ def __len__(self):
...
-CachingQuerySet = caching.base.CachingQuerySet
-CachingQuerySet.__bases__ = (TransformQuerySet,) + CachingQuerySet.__bases__
-
-
-class UncachedManagerBase(models.Manager):
+class ManagerBase(models.Manager):
+    # This needs to be set so that this manager class is being used
+    # for related objects too. This helps resolving translation fields
+    # on related fields.
+    use_for_related_fields = True
```
Comment (diox) [inline src/olympia/amo/models.py]: This also ensures we don't ignore soft-deleted items when traversing relations, if they are hidden by the `objects` manager, like we do with `Addon`.

Comment (EnTeQuAk) [reply src/olympia/amo/models.py]: I'll add that to the comment.

Comment (EnTeQuAk) [reply src/olympia/amo/models.py]: Added in 106b784f7e

**Pull request merged**